public/js/components/card-drop-down.jsx

```diff
@@ -39,40 +75,46 @@ export default class CardDropDown extends Component {
     this.setState({isFocused: false});
   }
 
-  handleChange = e => {
-    if (this.props.onCardChange) {
-      this.props.onCardChange(e);
-    }
-    var selectedText;
-    var selectedNode = e.target.options[e.target.selectedIndex];
-    if (selectedNode) {
-
-      var selectedCardType = selectedNode.getAttribute('data-type');
-      if (selectedCardType) {
-        selectedText = selectedNode.firstChild.nodeValue.replace(
-          selectedCardType.toUpperCase(), '');
-      }
-    }
+  updateCardSelection (selectedCardResource) {
+    var selectedCardData = this.getSelectedCard(selectedCardResource);
+    var selectedCardType = selectedCardData.type_name.toLowerCase();
+    var selectedText = '●●●● ●●●● ●●●● ' + selectedCardData.truncated_id;
     this.setState({
       selectedText: selectedText || defaultSelectText,
       selectedCardType: selectedCardType,
     });
   }
 
+  handleChange = e => {
+    if (this.props.onCardChange) {
+      this.props.onCardChange(e);
+    }
+    this.updateCardSelection(e.target.value);
+  }
+
   render() {
     var cardData = this.props.cards;
     var cardOptions = [];
 
-    cardOptions.push(<option value="">{defaultSelectText}</option>);
+    if (this.props.showDefaultOption === true) {
+      cardOptions.push((
+        <option
+          value=""
+          key="def"
+          ref="def">{defaultSelectText}</option>)
```

Comment:
nit: this trailing parenthesis looks off to me because it doesn't match the opening parentheses style. Only mentioning it because I stared at it a while thinking you'd forgotten a semi colon.